src/ports/linux/pnal_eth.c

```diff
@@ -142,6 +144,23 @@ pnal_eth_handle_t * pnal_eth_init (
    sll.sll_protocol = htons (linux_receive_type);
    bind (handle->socket, (struct sockaddr *)&sll, sizeof (sll));
 
+   /* Join profinet multicast group */
+   mreq.mr_ifindex = ifindex;
+   mreq.mr_type = PACKET_HOST | PACKET_MR_MULTICAST;
+   mreq.mr_alen = ETH_ALEN;
+   memcpy (mreq.mr_address, pn_mcast_addr, ETH_ALEN);
+
+   if (
+      setsockopt (
+         handle->socket,
+         SOL_PACKET,
+         PACKET_ADD_MEMBERSHIP,
+         &mreq,
+         sizeof (mreq)) != 0)
+   {
+      printf ("Failed to join Profinet multicast group\n");
```

Comment:
This should probably be a log message of some sort. Should the error be propagated to the caller?

Reply:
I added the log message as a PNAL message. All pnal files share the same tag.
Regarding the message I just wanted to keep this problem visible and then address it if it is a problem on other systems. 

Return codes for other ioctls or options are not checked in this function. It has worked without this additional socket option with previous linux systems we have been using and I do not think we need to propagate the error to user. 

I do not know why additional socket option is needed in this particular case. Just tried make it work.